Cython/Compiler/ExprNodes.py

```diff
@@ -1280,7 +1280,8 @@ def coerce_to(self, dst_type, env):
         return super().coerce_to(dst_type, env)
 
     def analyse_as_type(self, env):
-        return py_none_type
+        if not env.in_c_type_context:
+            return py_none_type
```

Comment:
If we return something from a function, I prefer *always* returning something, i.e. explicitly `return None` otherwise here, not just implicitly. If that case is simply missing, the code makes me wonder if that's intended or whether something else was forgotten, and that distracts me from reading the code.

Reply:
fixed.